src/System.Memory/tests/Performance/Perf.Span.IndexOf.cs

```diff
@@ -53,7 +53,7 @@ public void SpanIndexOfCharAsBytes(int size)
                 {
                     for (int i = 0; i < Benchmark.InnerIterationCount; i++)
                     {
-                        index |= byteSpan.IndexOf(53);        // '5' = 53
+                        index |= byteSpan.IndexOf<byte>(53);        // '5' = 53
```

Comment:
Why do you need all these <byte>? The span is already Span<byte> and so the T is byte.

Reply:
If I don't specify the type, it tries to treats '53' as an integer and tries to convert to byte and fails:
Span\IndexOf.byte.cs(17,34): error CS1503: Argument 2: cannot convert from 'int' to 'System.ReadOnlySpan<byte>'

I could either specify the type or cast the argument to byte. But one of those two approaches is necessary.
```C#
int idx = sp.IndexOf((byte)0); // works
int idx = sp.IndexOf<byte>(0); // works
int idx = sp.IndexOf(0); // fails
```

Reply:
I guess it makes sense, but it's bad :-(
cc: @jaredpar